storage/replica.go

```diff
@@ -448,65 +448,70 @@ func (r *Replica) requestLeaderLease(timestamp roachpb.Timestamp) <-chan *roachp
 		return llChan
 	}
 
-	r.store.Stopper().RunWorker(func() {
-		pErr := func() *roachpb.Error {
-			// TODO(Tobias): get duration from configuration, either as a config flag
-			// or, later, dynamically adjusted.
-			duration := DefaultLeaderLeaseDuration
-
-			// Prepare a Raft command to get a leader lease for this replica.
-			expiration := timestamp.Add(int64(duration), 0)
-			desc := r.Desc()
-			_, replica := desc.FindReplica(r.store.StoreID())
-			if replica == nil {
-				return roachpb.NewError(roachpb.NewRangeNotFoundError(r.RangeID))
-			}
-			args := &roachpb.LeaderLeaseRequest{
-				Span: roachpb.Span{
-					Key: desc.StartKey.AsRawKey(),
-				},
-				Lease: roachpb.Lease{
-					Start:      timestamp,
-					Expiration: expiration,
-					Replica:    *replica,
-				},
-			}
-			ba := roachpb.BatchRequest{}
-			ba.Timestamp = r.store.Clock().Now()
-			ba.RangeID = r.RangeID
-			ba.Add(args)
-
-			// Send lease request directly to raft in order to skip unnecessary
-			// checks from normal request machinery, (e.g. the command queue).
-			// Note that the command itself isn't traced, but usually the caller
-			// waiting for the result has an active Trace.
-			cmd, err := r.proposeRaftCommand(r.context(context.Background()), ba)
-			if err != nil {
-				return roachpb.NewError(err)
-			}
+	if !r.store.Stopper().RunTask(func() {
```

Comment:
I thought about adding a `MaybeRunWorker`, but this pattern is really a hack and not something we want to encourage. It looks like whatever problems you were running into with a task have been fixed (maybe by the context plumbing?), so I think we can just change it to a task: #5873